### PR TITLE
Make unused variable pattern configurable

### LINKED ISSUE #134: Make throwaway variable (`_`) pattern configurable
There are cases where `_` cannot be used to mark a throwaway variable and it would be great if ruff allowed configuring it. This is common in projects that use `gettext` for example, as `_` is used for translating strings there. One of the ways unused variables are marked in such projects is using `__` (double underscore) but it probably would be a good idea to just make it configurable with a sane default to appeal to more uses.

Notably, pylint allows configuring the dummy variable pattern with a regex:
https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#dummy-variables-rgx

### PR DESCRIPTION
**Pull request by harupy**
Resolve #134.

### CODE REVIEW

File: src/settings.rs [hunk 1]
```diff
@@ -101,6 +106,8 @@ static DEFAULT_EXCLUDE: Lazy<Vec<FilePattern>> = Lazy::new(|| {
     ]
 });
 
+static DEFAULT_DUMMY_VARIABLE_RGX: Lazy<Regex> = Lazy::new(|| Regex::new("^_+$").unwrap());
```
Comment (harupy) [inline src/settings.rs:109]: pylint uses the following default value:

```
_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_
```

https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#dummy-variables-rgx

File: src/settings.rs [hunk 2]
```diff
@@ -101,6 +106,9 @@ static DEFAULT_EXCLUDE: Lazy<Vec<FilePattern>> = Lazy::new(|| {
     ]
 });
 
+static DEFAULT_DUMMY_VARIABLE_RGX: Lazy<Regex> =
+    Lazy::new(|| Regex::new("_+$|_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$").unwrap());
```
Comment (harupy) [inline src/settings.rs:110]:
```suggestion
    Lazy::new(|| Regex::new("^_$").unwrap());
```

Should we align with flake8?

File: resources/test/fixtures/F841_dummy_variable_rgx.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+def f():
+    x = 0
+    _ = 0
+    __ = 0
+    _x = 0
+    __x = 0
+    x_ = 0
+    x__ = 0
+    __x__ = 0
+    _x_y_z = 0
```
Comment (harupy) [inline resources/test/fixtures/F841_dummy_variable_rgx.py:10]: Flake8 result:

```
> flake8 resources/test/fixtures/F841_dummy_variable_rgx.py
resources/test/fixtures/F841_dummy_variable_rgx.py:2:5: F841 local variable 'x' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:4:5: F841 local variable '__' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:5:5: F841 local variable '_x' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:6:5: F841 local variable '__x' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:7:5: F841 local variable 'x_' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:8:5: F841 local variable 'x__' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:9:5: F841 local variable '__x__' is assigned to but never used
resources/test/fixtures/F841_dummy_variable_rgx.py:10:5: F841 local variable '_x_y_z' is assigned to but never used
```

File: src/settings.rs [hunk 4]
```diff
@@ -102,6 +107,8 @@ static DEFAULT_EXCLUDE: Lazy<Vec<FilePattern>> = Lazy::new(|| {
     ]
 });
 
+static DEFAULT_DUMMY_VARIABLE_RGX: Lazy<Regex> = Lazy::new(|| Regex::new("^_$").unwrap());
```
Comment (charliermarsh) [inline src/settings.rs:110]: Should we make this `_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_` to match Pylint?

**Comment by charliermarsh** [comment]
Looks great!

File: src/settings.rs (hunk 4 above)
Comment (charliermarsh) [reply src/settings.rs:110]: (Will make this change then merge.)

Comment (harupy) [reply src/settings.rs:110]: Yes, that sounds good to me :)

Comment (charliermarsh) [reply src/settings.rs]: (A went for something in-between -- I think `dummy` and `ignored` and `unused` are a bit odd as defaults.)

Comment (harupy) [reply src/settings.rs]: Makes sense!

**Pull request merged**